app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,73 @@
+class PasswordResetsController < ApplicationController
+  before_action :load_user, only: [:edit, :update]
+  before_action :valid_user, only: [:edit, :update]
+  before_action :check_expiration, only: [:edit, :update]
+  before_action :validate_email, only: [:create]
+
+  # GET /password_resets/new
+  def new; end
+
+  # GET /password_resets/:id/edit
+  def edit; end
+
+  # POST /password_resets
+  # create password_reset, password_reset_digest, and send email
+  def create
+    @user.create_reset_digest
+    @user.send_password_reset_email
+    flash[:info] = t(".email_sent")
+    redirect_to root_path, status: :see_other
+  end
+
+  # PATCH /password_resets/:id
+  def update
+    if params[:user][:password].empty?
+      @user.errors.add(:password, t(".empty"))
+      render :edit, status: :unprocessable_entity
+    elsif @user.update(user_params)
```

Comment:
```suggestion
    elsif @user.update(user_params.merge(reset_digest: nil))
```
Update thêm mã reset_digest là nil